flink-core/src/main/java/org/apache/flink/core/memory/DataOutputSerializer.java

```diff
@@ -160,7 +160,6 @@ public void writeBytes(String s) throws IOException {
         for (int i = 0; i < sLen; i++) {
             writeByte(s.charAt(i));
         }
-        this.position += sLen;
```

Comment:
```java
    @Override
    public void write(int b) throws IOException {
        if (this.position >= this.buffer.length) {
            resize(1);
        }
        this.buffer[this.position++] = (byte) (b & 0xff); // here! position++;
    }
```
FYI, inside of `writeByte` it already increment position, so above `L163` updates position twice!